snapcraft/internal/project_loader/_config.py

```diff
@@ -308,3 +313,15 @@ def _expand_filesets_for(step, properties):
             new_step_set.append(item)
 
     return new_step_set
+
+
+def _mangle_yaml(snapcraft_yaml):
```

Comment:
I can't think of a good name here, but maybe we can improve mangle.
_add_sources_for_sourceless_plugins?